### PR TITLE
feat: add PATCH method on API resource

### PR DESCRIPTION
**Pull request by gcusnieux**
gravitee-io/issues#7443

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-uszxuyitxw.chromatic.com)

🚀 CI was able to deploy the build of this PR, so you can now try it directly [here](https://apimnightlywebui24386.blob.core.windows.net/7443-patch-import/index.html)
_Notes_: The deployed app is linked to the management API of APIM master. (Same login and password as APIM master)

### CODE REVIEW

File: gravitee-apim-rest-api/gravitee-apim-rest-api-model/src/main/java/io/gravitee/rest/api/model/JsonPatch.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class JsonPatch {
+
+    @JsonProperty(required = true)
+    String path;
+
+    Object value;
+
+    @JsonProperty(required = true)
```
Comment (brasseld) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-model/src/main/java/io/gravitee/rest/api/model/JsonPatch.java]: I think we should do Operation.REPLACE by default.

File: gravitee-apim-rest-api/gravitee-apim-rest-api-model/src/main/java/io/gravitee/rest/api/model/JsonPatch.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+ * limitations under the License.
+ */
+package io.gravitee.rest.api.model;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class JsonPatch {
+
+    @JsonProperty(required = true)
+    String path;
```
Comment (brasseld) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-model/src/main/java/io/gravitee/rest/api/model/JsonPatch.java]: Could it be more explicit like `jsonPath` ?

**Review by brasseld — COMMENTED** [review]
Cool stuff, I like it!

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/PlanServiceImpl.java [hunk 3]
```diff
@@ -727,6 +727,10 @@ private void assertPlanSecurityIsAllowed(final ExecutionContext executionContext
     @Override
     public boolean anyPlanMismatchWithApi(List<String> planIds, String apiId) {
         try {
+            Set<Plan> plans = planRepository.findByApi(apiId);
+            if (!plans.isEmpty() && planIds.isEmpty() || planIds.size() < plans.size()) {
```
Comment (a-cordier) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/PlanServiceImpl.java]: If we introduce this change we have to keep in mind that the doc needs to be updated

https://docs.gravitee.io/apim/3.x/apim_publisherguide_import_apis.html#plans

**Review by a-cordier — APPROVED** [review]

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/test/java/io/gravitee/rest/api/service/impl/PlanService_MismatchWithApiTest.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    private PlanService planService = new PlanServiceImpl();
+
+    @Mock
+    private PlanRepository planRepository;
+
+    @Test
+    public void shouldNotMismatchWithApi() throws TechnicalException {
+        Plan plan = mock(Plan.class);
+        when(plan.getApi()).thenReturn(API_ID);
+        when(planRepository.findByApi(API_ID)).thenReturn(Set.of(plan));
```
Comment (a-cordier) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/test/java/io/gravitee/rest/api/service/impl/PlanService_MismatchWithApiTest.java]:
```suggestion

```

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/test/java/io/gravitee/rest/api/service/impl/PlanService_MismatchWithApiTest.java [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+        when(planRepository.findByIdIn(List.of(PLAN_ID))).thenReturn(Set.of(plan));
+
+        assertFalse(planService.anyPlanMismatchWithApi(List.of(PLAN_ID), API_ID));
+    }
+
+    @Test
+    public void shouldMismatchWithApi() throws TechnicalException {
+        Plan plan = mock(Plan.class);
+        when(plan.getApi()).thenReturn("OTHER_API");
+        when(planRepository.findByApi(API_ID)).thenReturn(Set.of(plan));
```
Comment (a-cordier) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/test/java/io/gravitee/rest/api/service/impl/PlanService_MismatchWithApiTest.java]:
```suggestion
 
```

**Review by a-cordier — CHANGES_REQUESTED** [review]

**Review by a-cordier — APPROVED** [review]

**Pull request merged**